lib/import/colorectal/providers/leeds/leeds_handler_colorectal.rb

```diff
@@ -1,648 +1,403 @@
+                          rep.scan(EXON_VARIANT_REGEX).size.positive?
+
+              pos_genes -= [gene]
+              varclass = gene_path_hash[gene] || @var_class
+              process_abnormal_gene(rep, genocolorectal, genotypes, gene, varclass)
+            end
+
+            return if pos_genes.blank?
+
+            process_negative_genes(pos_genes, genocolorectal, genotypes)
+          end
+
+          def process_abnormal_gene(rep, genocolorectal, genotypes, gene, varclass)
+            genocolorectal_dup = genocolorectal.dup_colo
+            process_cdna_variant(genocolorectal_dup, rep)
+            process_exonic_variant(genocolorectal_dup, rep)
+            process_protein_impact(genocolorectal_dup, rep)
+            genocolorectal_dup.add_gene_colorectal(gene)
+            genocolorectal_dup.add_variant_class(varclass)
+            @teststatus = 10 if NON_PATH_VARCLASS.include? varclass
+            genocolorectal_dup.add_status(@teststatus)
+            genotypes.append(genocolorectal_dup)
+          end
+
+          def process_negative_genes(negative_genes, genocolorectal, genotypes)
+            negative_genes&.each do |gene|
+              genocolorectal_dup = genocolorectal.dup_colo
+              genocolorectal_dup.add_gene_colorectal(gene)
+              genocolorectal_dup.add_status(1)
+              genotypes << genocolorectal_dup
+            end
+          end
+
+          def allocate_genes(_record)
+            @genes_panel = get_genes_from_report(@report)
+
+            return if @genes_panel.present?
+
+            GENES_PANEL.each do |panel, genes|
+              @genes_panel = genes if @genes_hash[panel].include?(@geno)
+            end
+          end
+
+          def find_test_status(_record)
+            @teststatus = nil
+            if @geno == 'ngs msh2 seq variant'
+              @teststatus = @report.include?('likely to be benign') ? 10 : 2
+            else
+              STATUS_PANEL.each do |category, status|
+                @teststatus = status if @status_hash[category].include?(@geno)
+              end
+            end
+            # exceptional cases for 'mlh1 only -ve' genotype
+            exceptinal_teststatus if @geno == 'mlh1 only -ve'
+            @teststatus
+          end
+
+          def exceptinal_teststatus
+            if @report.match(/insufficient\sDNA|DNA\sprovided\sis\sof\sinsufficient/i)
+              @teststatus = 9
+            elsif @report.include?('The variant c.1409+47T>C was detected in intron 12')
+              @teststatus = 10
+            elsif @report.include?('sequence variant c.1595G>A')
+              @teststatus = 2
+            end
+          end
+
+          def cleanup_report(report)
+            raw_report = report
+            exclude_statements = [
+              'Screening for mutations in MLH1, MSH2 and MSH6 is now in progress as requested.',
+              'MLPA and MSH2 analysis was not requested.',
```

Comment:
Would it be good to add these to the constants file instead?

Reply:
Done in latest commit .